src/Microsoft.HttpRepl/Commands/TreeNode.cs

```diff
@@ -30,7 +32,7 @@ private TreeNode(Formatter formatter, string prefix, string entry, int depth)
 
         public TreeNode AddChild(string prefix, string entry)
         {
-            TreeNode child = new TreeNode(_formatter, prefix, entry, _depth + 1);
+            TreeNode child = new TreeNode(_formatter, prefix ?? throw new ArgumentNullException(nameof(prefix)), entry, _depth + 1);
```

Comment:
it feels weird to embed the null check in the constructor arguments.  For readability, I'd've expected the null check first, then the call to the constructor.  Also, in tandem with the other comment here, this null check seems like it belongs inside the private constructor, not inline in the call.

Reply:
Same answer as above.